source/agora/node/Validator.d

```diff
@@ -275,7 +275,8 @@ public class Validator : FullNode, API
     protected override void discoveryTask ()
     {
         this.network.discover(this.required_peer_utxos);
-        this.timers[TimersIdx.Discovery].rearm(this.config.node.network_discovery_interval, false);
+        if (!this.is_shutting_down)
+            this.timers[TimersIdx.Discovery].rearm(this.config.node.network_discovery_interval, false);
```

Comment:
Can't we use `startTaskTimer` of the FullNode by changing the function to `protected`?

Reply:
Yes good idea. I'll update.